staking/cli/src/instructions.rs

```diff
@@ -0,0 +1,472 @@
+    let transaction_signature_res = rpc_client
+        .send_and_confirm_transaction_with_spinner_and_config(
+            &transaction,
+            CommitmentConfig::confirmed(),
+            RpcSendTransactionConfig {
+                skip_preflight: true,
+                ..Default::default()
+            },
+        );
+    match transaction_signature_res {
+        Ok(signature) => {
+            println!("Transaction successful : {signature:?}");
+        }
+        Err(err) => {
+            println!("transaction err: {err:?}");
+        }
+    }
+}
+
+pub fn process_write_encoded_vaa(
+    rpc_client: &RpcClient,
+    vaa: &[u8],
+    wormhole: Pubkey,
+    payer: &Keypair,
+) -> Pubkey {
+    let encoded_vaa_keypair = Keypair::new();
+    let encoded_vaa_size: usize = vaa.len() + VAA_START;
+
+    let create_encoded_vaa = system_instruction::create_account(
+        &payer.pubkey(),
+        &encoded_vaa_keypair.pubkey(),
+        Rent::default().minimum_balance(encoded_vaa_size),
+        encoded_vaa_size as u64,
+        &wormhole,
+    );
+    let init_encoded_vaa_accounts = wormhole_core_bridge_solana::accounts::InitEncodedVaa {
+        write_authority: payer.pubkey(),
+        encoded_vaa:     encoded_vaa_keypair.pubkey(),
+    }
+    .to_account_metas(None);
+
+    let init_encoded_vaa_instruction = Instruction {
+        program_id: wormhole,
+        accounts:   init_encoded_vaa_accounts,
+        data:       wormhole_core_bridge_solana::instruction::InitEncodedVaa.data(),
+    };
+
+    let write_encoded_vaa_accounts = wormhole_core_bridge_solana::accounts::WriteEncodedVaa {
+        write_authority: payer.pubkey(),
+        draft_vaa:       encoded_vaa_keypair.pubkey(),
+    }
+    .to_account_metas(None);
+
+    let write_encoded_vaa_accounts_instruction = Instruction {
+        program_id: wormhole,
+        accounts:   write_encoded_vaa_accounts.clone(),
+        data:       wormhole_core_bridge_solana::instruction::WriteEncodedVaa {
+            args: WriteEncodedVaaArgs {
+                index: 0,
+                data:  vaa.to_vec(),
+            },
+        }
+        .data(),
+    };
+
+    let (header, _): (Header, Body<&RawMessage>) = serde_wormhole::from_slice(vaa).unwrap();
+    let guardian_set = GuardianSet::key(&wormhole, header.guardian_set_index);
+
+    let request_compute_units_instruction: Instruction =
+        ComputeBudgetInstruction::set_compute_unit_limit(600_000);
+
+    let verify_encoded_vaa_accounts = wormhole_core_bridge_solana::accounts::VerifyEncodedVaaV1 {
+        guardian_set,
+        write_authority: payer.pubkey(),
+        draft_vaa: encoded_vaa_keypair.pubkey(),
+    }
+    .to_account_metas(None);
+
+    let verify_encoded_vaa_instruction = Instruction {
+        program_id: wormhole,
+        accounts:   verify_encoded_vaa_accounts,
+        data:       wormhole_core_bridge_solana::instruction::VerifyEncodedVaaV1 {}.data(),
+    };
+
+    process_transaction(
```

Comment:
note: on mainnet, the vaa is bigger so we'll need to do this in multiple transactions like in `write_publisher_caps`, but we can fix that once we need it

Reply:
sounds good.